### PR TITLE
[CENNSO-105]Trigger Usage Report for removed URR

### PR DESCRIPTION
**Pull request by sergeymatov**
If URR is instructed to be removed by PFCP Session Modification Request UP-F should include Usage Report for such URR either into PFCP Session Modification Response or separate Session Usage Report.

### CODE REVIEW

**Review by ivan4th — CHANGES_REQUESTED** [review]

File: test/e2e/upg_e2e.go [hunk 1]
```diff
@@ -992,6 +992,51 @@ var _ = ginkgo.Describe("[Reporting]", func() {
...
+		f := framework.NewDefaultFramework(framework.UPGModeTDF, framework.UPGIPModeV4)
+		ginkgo.It("should generate usage report", func() {
+			ginkgo.By("Removing URRs from session should trigger Usage Report")
+			sessionCfg := &framework.SessionConfig{
+				IdBase:      1,
+				UEIP:        f.UEIP(),
+				Mode:        f.Mode,
+				VolumeQuota: 100000,
+				NoURRs:      false,
+				// request a report _after_ the monitoring time
```
[the remark this answers is not in the document]
Comment (sergeymatov) [reply test/e2e/upg_e2e.go:1005]: Sorry, copied from another test case...

[the remark this answers is not in the document]
Comment (sergeymatov) [reply test/e2e/upg_e2e.go]: fixed)

File: test/e2e/framework/sessionconfig.go [hunk 2]
```diff
@@ -336,12 +336,12 @@ func (cfg SessionConfig) CreateOrUpdateURR(id uint32, update bool) *ie.IE {
 	return urr
 }
 
-func (cfg SessionConfig) CreateVolumeURR(id uint32) *ie.IE {
+func (cfg SessionConfig) CreateVolumeURR(id uint32, tvol uint32) *ie.IE {
```
[the remark this answers is not in the document]
Comment (sergeymatov) [reply test/e2e/framework/sessionconfig.go]: done

File: upf/upf_pfcp_api.c [hunk 3]
```diff
@@ -2644,6 +2645,34 @@ handle_session_modification_request (pfcp_msg_t * msg,
 
   resp_dmsg.seid = sess->cp_seid;
 
+  active = pfcp_get_rules (sess, PFCP_ACTIVE);
+  upf_usage_report_init (&report, vec_len (active->urr));
```
[the remark this answers is not in the document]
Comment (sergeymatov) [reply upf/upf_pfcp_api.c]: done

**Review by ivan4th — APPROVED** [review]
LGTM

File: upf/upf_pfcp_api.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (RoadRunnr) [reply upf/upf_pfcp_api.c]: the solution with `have_report` is not very elegant.
You could have added the number of possible report to the `upf_usage_report_t`, set the `reports` and  `liusa` fields to `NULL` and alloc them when needed in `upf_usage_report_trigger`

**Pull request merged**